suprb/optimizer/rule/ns/archive.py

```diff
@@ -0,0 +1,53 @@
+from abc import ABCMeta, abstractmethod
+
+from suprb.rule import Rule
+from suprb.base import BaseComponent
+from suprb.utils import RandomState
+
+
+class Archive(BaseComponent, metaclass=ABCMeta):
+    """Base Archive Class to store the rules from previous populations"""
+
+    def __init__(self):
+        self.archive = []
+
+    def set_archive(self, pool: list[Rule]):
```

Comment:
while the default is to set it to the pool, we don't need to reflect this in the name. `rule_set` would be more appropriate

Reply:
also, use a property?